dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityAttributeStore.java

```diff
@@ -78,6 +78,20 @@
      */
     Set<TrackedEntityAttribute> getTrackedEntityAttributesByTrackedEntityTypes();
 
+    /**
+     * Retrieve all attributes that are either configured as searchable in any
+     * program or tracked entity type or are configured to be unique.
+     *
+     * @return set of TrackedEntityAttribute
+     */
+    Set<TrackedEntityAttribute> getAllSearchableAndUniqueTrackedEntityAttributes();
+
+    boolean createTrigramIndexForAttribute( TrackedEntityAttribute trackedEntityAttribute );
+
+    boolean runAnalyzeOnTrackerTables();
```

Comment:
My thought process was to make the naming kinda future-proof. Since its also the persistence layer method I thought I could give it more context. The implementation inside might still change (as of now its is just one table), but it could be expanded to some more related tracker tables. So I thought a generic name would be best instead of just "runAnalyze" which might make it seem like its only tun on TrackedEntityAttribute table (following the store name).